### ISSUE TITLE
Encrypted images - 'NoneType' object has no attribute 'getAttributeValue' - log comparison

### ISSUE THREAD

**Issue by jodersus** [tgalal/yowsup#1391]
Hi,

Following what I commented in #1389, I have checked logs of how images were received a few days ago, and how we're getting them now. I reckon the issue is that xml tags have changed.

@jlguardi I'm posting them here to see if it helps fix the `'NoneType' object has no attribute 'getAttributeValue'` problem; I'm a bit over my head right now.

```
22/03/2016 04:59:17 PM - yowsup.layers.logger.layer - DEBUG - rx:
<message participant="XXXXXXXXX@s.whatsapp.net" notify="El Nikel" from="XXXXXXXXX-XXXXXXXXXX@g.us" type="media" id="FA85CADF9CE9BE0117" t="1458665959">
<media mimetype="image/jpeg" filehash="pZq5yDVbKBKDQt2MOjh/qE+sRE2nxfG5afJrbFokxyE=" encoding="raw" url="https://mmi685.whatsapp.net/d/ioEQWyGzo4vBiQDWLvmvPFbxeecABS6mIlXsrA/Aj9ArO9tC6olRQM21LQ-7_image.jpg" ip="x.x.x.x" height="278" caption="caption" file="Aj9ArO9tC6olRQM21LQ-7_KsSkaKok6g5jRN9I-8K_kV.jpg" type="image" width="320" size="17671">
--> CONTENTS <--
HEX:ffd8.....
</media>
</message>

23/03/2016 12:37:17 PM
DEBUG:yowsup.layers.logger.layer:rx:
<message type="media" notify="Alex M" from="XXXXXXXXX@s.whatsapp.net" id="39937A0E111A52A78023" t="1458735747">
<enc count="1" oid="39937A0E111A52A78023" mediatype="image" v="2" ot="1458731337" type="pkmsg">
--> ENCRYPTED MESSAGE <--
HEX:3308f...
</enc>
</message>
```

**Comment by jlguardi** [comment]
Image/media encryption isn't supported yet. Use s40 env instead.

**Comment by yniv** [comment]
@jlguardi but we're getting this error in the s40 env!

**Comment by jodersus** [comment]
I'm using s40 env

**Comment by jodersus** [comment]
I am using s40 env, as I was using before.

**Comment by jlguardi** [comment]
Have you started a chat with this recipient using android and now you have changed to S40? So skip (ack) this message (is encrypted but now you don't support) and next could work

**Comment by jlguardi** [comment]
@jodersus Fuck! Now I'm lost!!

**Comment by jlguardi** [comment]
Could you try the retry posted in #1389?

**Comment by jodersus** [comment]
updating the code now, gimme a minute...

**Comment by jodersus** [comment]
@jlguardi creo que es una tontería como que antes el atributo del xml se llama "type" dentro de "media" y ahora está en "enc" con el nombre "mediatype"

Sorry for Spanish...

**Comment by yniv** [comment]
@jodersus @jlguardi i think it's more than that because the content is now encrypted (i used google translate :) )

**Comment by jodersus** [comment]
:_)

I'm sure encryption is going to be more complicated, but the first issue is that yowsup tries to get a child attribute that is not there (because it has a different name). 
That's my guess, the error says NoneType and "no attribute", so....
and I reckon if we clear that, we'll get to the next issue...

Cheers :)

**Comment by yniv** [comment]
i think @mgp25 's php chat-api is supporting this. maybe someone can port it?

**Comment by yniv** [comment]
@tgalal @jlguardi @jodersus 
does someone is working on this? or have any news ?

**Comment by yniv** [comment]
i just seen that @tgalal said he's working on it in the wa16 branch... maybe we can look into that too

**Comment by jodersus** [comment]
I am working on this issue (this post) not the encryption on a wider level (and I could not, no time)

**Comment by yniv** [comment]
have someone checked the wa16 branch??? it looks @tgalal has committed changes regarding this.
maybe it works??

**Comment by yniv** [comment]
@jodersus pls send your fix when you have it

**Comment by jodersus** [comment]
Of course!
I am trying the approach I described here, but I get more similar errors in different parts of the code.
I am not that familiar with yowsup detail, and it is taking me too long, but if I ever get anywhere working, I will post it here, no doubt.

**Comment by yniv** [comment]
fix to ignore media messages for now:
```
 def recvMessageStanza(self, node):
        if node.getAttributeValue("type") == "media":
            mediaNode = node.getChild("media")

	    if mediaNode == None:
                return None

            if mediaNode.getAttributeValue("type") == "image":
                entity = ImageDownloadableMediaMessageProtocolEntity.fromProtocolTreeNode(node)
                self.toUpper(entity)
            elif mediaNode.getAttributeValue("type") == "audio":
                entity = AudioDownloadableMediaMessageProtocolEntity.fromProtocolTreeNode(node)
                self.toUpper(entity)
            elif mediaNode.getAttributeValue("type") == "video":
                entity = VideoDownloadableMediaMessageProtocolEntity.fromProtocolTreeNode(node)
                self.toUpper(entity)
            elif mediaNode.getAttributeValue("type") == "location":
                entity = LocationMediaMessageProtocolEntity.fromProtocolTreeNode(node)
                self.toUpper(entity)
            elif mediaNode.getAttributeValue("type") == "vcard":
                entity = VCardMediaMessageProtocolEntity.fromProtocolTreeNode(node)
                self.toUpper(entity)
```

**Comment by chriso0710** [comment]
I don't see any commits on the WA16 branch. Last commit was on Feb 15.

**Comment by jodersus** [comment]
@yniv where did you put that code? which layer?

**Comment by eduardobento** [comment]
In my tests, WA16 branch receive encrypted private images...
But messages (Text or media) with type="msg" does not...

**Comment by bdbais** [comment]
File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.4.48-py2.7.egg/yowsup/layers/__init__.py", line 105, in receive
    recv(node)
  File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.4.48-py2.7.egg/yowsup/layers/protocol_media/layer.py", line 40, in recvMessageStanza
    if mediaNode.getAttributeValue("type") == "image":
AttributeError: 'NoneType' object has no attribute 'getAttributeValue'

tried lastest @jlguardi encrypted branch but nothing changed.

**Comment by jlguardi** [comment]
Please, attach logs!

De: bdbais
Enviado: domingo, 27 de marzo de 2016 21:04
Para: tgalal/yowsup
Responder a: tgalal/yowsup
CC: jlguardi
Asunto: Re: [yowsup] Encrypted images - 'NoneType' object has no attribute 'getAttributeValue' - log comparison (#1391)


  File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.4.48-py2.7.egg/yowsup/layers/__init__.py", line 105, in receive
    recv(node)
  File "/usr/local/lib/python2.7/dist-packages/yowsup2-2.4.48-py2.7.egg/yowsup/layers/protocol_media/layer.py", line 40, in recvMessageStanza
    if mediaNode.getAttributeValue("type") == "image":
AttributeError: 'NoneType' object has no attribute 'getAttributeValue'

tried lastest @jlguardi encrypted branch but nothing changed.

---

**Comment by jodersus** [comment]
@bdbais I think you need to *re-install* the whole thing from scratch, out of @jlguardi master branch.

**Comment by tgalal** [comment]
fixed in v3.2.0
